### PR TITLE
Introduce support for S32K344

### PR DESCRIPTION
**Pull request by manuargue**
Add a sub-set of S32K3 baremetal drivers (NXP Real-Time Drivers), SoC-specific headers and pinmux headers to enable support for S32K344 devices. The S32K3 family share some fair amount of peripherals with other non-S32 devices so in order to reuse existing shim drivers in Zephyr, allow to build MCUX drivers for this SoC as well. MCUX doesn't have official support for S32K3 so some glue code/headers is necessary to expose the device features and peripheral registers to MCUX.

### CODE REVIEW

**Comment by manuargue** [comment]
Hi @dleach02, could you please have a look to these changes and assign reviewers as needed? Thanks

**Comment by mmahadevan108** [comment]
@manuargue Is the approach to create a SoC specific folders under drivers. Are there any drivers are common across SoC's.

**Comment by manuargue** [comment]
> @manuargue Is the approach to create a SoC specific folders under drivers. Are there any drivers are common across SoC's.

Unfortunately the baremetal drivers are released independently per platform (release code is stripped based definitions), so there is no simple way to reuse them.

**Review by mmahadevan108 — APPROVED** [review]

**Comment by igalloway** [comment]
@dleach02  @danieldegrasse  - any chance of getting this approved soon? We are waiting on this to help with the additional drivers.

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,17 +2,16 @@ if(CONFIG_HAS_MCUX OR CONFIG_HAS_IMX_HAL OR CONFIG_HAS_S32_HAL)
   zephyr_library()
 endif()
 
-add_subdirectory_ifdef(
-  CONFIG_HAS_MCUX
-  mcux
-  )
+if(CONFIG_HAS_S32_HAL)
+  add_subdirectory(s32)
```
Comment (dleach02) [inline CMakeLists.txt:6]: why did you have to structure things this way?

Comment (manuargue) [reply CMakeLists.txt:6]: Right. When using MCUX with S32, MCUX sources are built from the ./s32/CMakeLists.txt. We don't want to include also the ./mcux/CMakeLists.txt. This allow to build S32 with `CONFIG_HAS_S32_HAL` and `CONFIG_HAS_MCUX`.

**Comment by manuargue** [comment]
Hi @dleach02 , could you please revisit this pr and if all comments have been resolved the consider to merge it?
We are looking forward to continue pushing changes for this board, and this first pr is blocking the rest of the support. Thanks

**Pull request merged**